src/databricks/labs/remorph/helpers/validate.py

```diff
@@ -71,46 +62,59 @@ def validate_format_result(self, config: MorphConfig, input_sql: str):
 
         return result, exception
 
-    # [TODO] Implement Debugger Logger
-    def query(self, query: str, catalog_name=None, schema_name=None):
+    def query(self, sql_backend: SqlBackend, query: str):
         """
         Validate a given SQL query using the Spark session.
 
         Parameters:
         - query (str): The SQL query to be validated.
-        - catalog_nm (str): The catalog name for the Spark session.
-        - schema_nm (str): The schema name for the Spark session.
+        - sql_backend (SqlBackend): The SQL backend to be used for validation.
 
         Returns:
         - tuple: A tuple containing a boolean indicating whether the query is valid or not,
         and a string containing a success message or an exception message.
         """
-        spark = self.spark
-
+        # When variables is mentioned Explain fails we need way to replace them before explain is executed.
+        explain_query = f'EXPLAIN {query.replace("${", "`{").replace("}", "}`").replace("``", "`")}'
         try:
-            # [TODO]: Explain needs to redirected to different console
-            # [TODO]: Hacky way to replace variables representation
-            self.use_schema(catalog_name, schema_name)
-            # When variables is mentioned Explain fails we need way to replace them before explain is executed.
-            spark.sql(query.replace("${", "`{").replace("}", "}`").replace("``", "`")).explain(True)
+            sql_backend.execute(explain_query)
             return True, None
-        except ParseException as pre:
-            logger.debug(f"Syntax Exception : NOT IGNORED. Flag as syntax error: {str(pre)}")
-            return False, str(pre)
-        except AnalysisException as aex:
-            if "[TABLE_OR_VIEW_NOT_FOUND]" in str(aex):
-                logger.debug(f"Analysis Exception : IGNORED: {str(aex)}")
-                return True, str(aex)
-            if "[TABLE_OR_VIEW_ALREADY_EXISTS]" in str(aex):
-                logger.debug(f"Analysis Exception : IGNORED: {str(aex)}")
-                return True, str(aex)
-            if "[UNRESOLVED_ROUTINE]" in str(aex):
-                logger.debug(f"Analysis Exception : NOT IGNORED: Flag as Function Missing error {str(aex)}")
-                return False, str(aex)
-
-            if "Hive support is required to CREATE Hive TABLE (AS SELECT).;" in str(aex):
-                logger.debug(f"Analysis Exception : IGNORED: {str(aex)}")
-                return True, str(aex)
-
-            logger.debug(f"Unknown Exception: {str(aex)}")
-            return False, str(aex)
+        except DatabricksError as dbe:
+            err_msg = str(dbe)
+            if "[PARSE_SYNTAX_ERROR]" in err_msg:
```

Comment:
Most of these exceptions are already concrete types in LSQL, refactor to multiple except clauses. Add more mapping to lsql if necessary